File: ios/GuideStreamTV/BrandWordmark.swift
//
//  BrandWordmark.swift
//  GuideStreamTV
//

import SwiftUI

public struct BrandWordmark: View {
    public enum WordmarkSize {
        case large
        case nav
        case small

        var baseSize: CGFloat {
            switch self {
            case .large: return 36
            case .nav: return 18
            case .small: return 13
            }
        }

        var tvSize: CGFloat {
            switch self {
            case .large: return 20
            case .nav: return 11
            case .small: return 8
            }
        }

        var weight: Font.Weight {
            switch self {
            case .large: return .heavy
            case .nav: return .bold
            case .small: return .semibold
            }
        }
    }

    var wordmarkSize: WordmarkSize = .nav

    public var body: some View {
        HStack(alignment: .firstTextBaseline, spacing: 0) {
            Text("Guide")
                .scaledFont(size: wordmarkSize.baseSize, weight: wordmarkSize.weight)
                .foregroundStyle(.white)
            Text("Stream")
                .scaledFont(size: wordmarkSize.baseSize, weight: wordmarkSize.weight)
                .foregroundStyle(Color(red: 0xF5 / 255, green: 0x82 / 255, blue: 0x1F / 255))
            Text(" TV")
                .scaledFont(size: wordmarkSize.tvSize, weight: wordmarkSize.weight)
                .foregroundStyle(Color(red: 0x5B / 255, green: 0xB0 / 255, blue: 0xFF / 255))
        }
        .accessibilityElement(children: .combine)
        .accessibilityLabel("GuideStream TV")
    }
}

File: android/app/src/main/java/com/rork/guidestreamtvandroid/ui/theme/BrandWordmark.kt
package com.rork.guidestreamtvandroid.ui.theme

import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.BaselineShift
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

enum class WordmarkSize(val baseSize: Int, val tvSize: Int, val weight: FontWeight) {
    LARGE(36, 20, FontWeight.Black),
    NAV(20, 12, FontWeight.Black),
    SMALL(13, 8, FontWeight.Bold),
}

/**
 * "GuideStream TV" wordmark.
 * Guide (white) + Stream (orange) + TV (light-blue, superscript).
 * Mirrors iOS BrandWordmark.swift.
 */
@Composable
fun BrandWordmark(
    size: WordmarkSize = WordmarkSize.NAV,
    modifier: Modifier = Modifier,
) {
    Row(
        modifier = modifier,
        verticalAlignment = Alignment.Bottom,
    ) {
        Text(
            text = "Guide",
            fontSize = size.baseSize.sp,
            fontWeight = size.weight,
            color = TextPrimary,
        )
        Text(
            text = "Stream",
            fontSize = size.baseSize.sp,
            fontWeight = size.weight,
            color = BrandOrange,
        )
        Text(
            text = "TV",
            fontSize = size.tvSize.sp,
            fontWeight = FontWeight.Bold,
            color = LightBlue,
            baselineShift = BaselineShift(0.3f),
            modifier = Modifier.padding(start = 2.dp),
        )
    }
}
